lib/routes/auth.js

```diff
@@ -1,25 +1,55 @@
 const router = require('express').Router();
+const bcrypt = require('bcryptjs');
+const jwt = require('jsonwebtoken');
 const User = require('../models/User');
+const { validateRegistration, validateLogin } = require('../validation');
 
 router.post('/register', async (request, response) => {
+  const { error } = validateRegistration(request.body);
+
+  if (error) return response.status(403).send(error.message);
+
+  const emailCheck = await User.findOne({ email: request.body.email });
+  const usernameCheck = await User.findOne({ username: request.body.username });
+  if (emailCheck) return response.status(400).send('E-mail already exists');
+  if (usernameCheck)
+    return response.status(400).send('This username is already taken');
+
+  const salt = await bcrypt.genSalt(10);
+  const hashedPassword = await bcrypt.hash(request.body.password, salt);
+
+  const user = new User({
+    username: request.body.username,
+    email: request.body.email,
+    password: hashedPassword,
+  });
+
   try {
```

Comment:
This `try` position will only capture errors on user save. 
But there are many more possibilities which should be caught.